packages/core/src/external/carequality/ihe-gateway-v2/report/report.ts

```diff
@@ -0,0 +1,301 @@
+import { QueryTypes, Sequelize } from "sequelize";
+import dayjs from "dayjs";
+import {
+  OutboundPatientDiscoveryResp,
+  XCPDGateway,
+  OutboundDocumentQueryResp,
+  XCAGateway,
+  OutboundDocumentRetrievalResp,
+} from "@metriport/ihe-gateway-sdk";
+
+const patientDiscoveryReportRequests = 10;
+const documentQueryReportRequests = 100;
+const documentRetrievalReportRequests = 100;
+
+type Report = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+  errorOids: {
+    oid: string;
+    error: string;
+  }[];
+};
+
+type SimplifiedReport = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+};
+
+type ReportSummary = {
+  simplifiedReports: SimplifiedReport[];
+  averageSuccessPercentage: number;
+  averageFailurePercentage: number;
+  errorDetailsMap: { [key: string]: { count: number; details: string } };
+};
+
+async function getRandomRequestIds({
+  sequelize,
+  tableName,
+  since,
+  limit,
+}: {
+  sequelize: Sequelize;
+  tableName: string;
+  since: string;
+  limit: number;
+}): Promise<string[]> {
+  const query = `
+      SELECT request_id
+      FROM ${tableName}
+      WHERE created_at >= '${since}'
+      ORDER BY created_at DESC
+      LIMIT ${limit};
+    `;
+  const results = await sequelize.query(query, { type: QueryTypes.SELECT });
+  // eslint-disable-next-line
+  return results.map((row: any) => row.request_id);
+}
+
+// execute asynch s
+async function getPatientDiscoveryResultsForRequestId({
+  sequelize,
+  requestId,
+}: {
+  sequelize: Sequelize;
+  requestId: string;
+}): Promise<OutboundPatientDiscoveryResp[]> {
+  const query = `
+        SELECT *
+        FROM patient_discovery_result
+        WHERE request_id = '${requestId}';
+    `;
+  const results = await sequelize.query(query, { type: QueryTypes.SELECT });
+  return results.map(result => {
+    //eslint-disable-next-line
+    return (result as any).data as OutboundPatientDiscoveryResp;
+  });
+}
+
+async function getDocumentQueryResultsForRequestId({
+  sequelize,
+  requestId,
+}: {
+  sequelize: Sequelize;
+  requestId: string;
+}): Promise<OutboundDocumentQueryResp[]> {
+  const query = `
+    SELECT *
+    FROM document_query_result
+    WHERE request_id = '${requestId}';
```

Comment:
ditto: have table name and column names syntactically tied to the TS types/model.

Reply:
The original types and model are defined in the api so I can't access those directly. I defined them in enums in the file though so at least they arent hardcoded in